### PR TITLE
speed up computations + show progress bar

### PR DESCRIPTION
**Pull request by pverscha**
This PR provides support for multicore CPU's and improves the computation time of the analysis. A new progress bar that displays the current analysis progress in the web application has also been added:

image

The lin metric PR has been merged into this PR.

### CODE REVIEW

File: api/app.py [hunk 1]
```diff
@@ -20,21 +26,51 @@ def analyze():
...
-        },
-        "invalid": list(not_present)
+        "analysis_id": id
     }
 
 
+@app.route('/progress/<id>', methods=["POST"])
+@cross_origin()
+def progress(id):
+    print(RESULTS)
```
Comment (rababerladuseladim) [inline api/app.py:42]: is this print statement a debugging artifact?

File: megago/megago.py [hunk 2]
```diff
@@ -174,7 +176,7 @@ def get_default_go_dag():
     return GODag(GO_DAG_FILE_PATH, prt=open(os.devnull, 'w'))
 
 
-def run_comparison(go_list_1, go_list_2, go_dag=None):
+def run_comparison(go_list_1, go_list_2, go_dag=None, progress=None):
```
Comment (rababerladuseladim) [inline megago/megago.py:179]: could you add the progress parameter to the docstring? I do not understand what kind of parameter is expected. Is it a function?

**Review by rababerladuseladim — CHANGES_REQUESTED** [review]
Looks great, only minor comments from my side.

File: api/app.py (hunk 1 above)
Comment (pverscha) [reply api/app.py]: This is indeed a debugging artificat :) I removed it

File: megago/megago.py (hunk 2 above)
Comment (pverscha) [reply megago/megago.py:179]: Done :)

**Review by rababerladuseladim — APPROVED** [review]

**Comment by rababerladuseladim** [comment]
please merge ahead!

**Pull request merged**